### PR TITLE
Fix for the EnumerationSerializer

### LINKED ISSUE #75: Concurrent deserialization of Collections of Enums leads to nulls and duplicated Enumeration#Values
Test case:

```
  case class Bar(time:Time)

  "Enumeration deserialization" should "be threadsafe" in {
    import scala.concurrent.ExecutionContext.Implicits.global

    val defaultConfig = ConfigFactory.parseString("""
    akka {
      extensions = ["com.romix.akka.serialization.kryo.KryoSerializationExtension$"]
      actor {
        serializers {
          kryo         = "com.romix.akka.serialization.kryo.KryoSerializer"
        }
        serialization-bindings {
          "java.io.Serializable"        = kryo
        }
        kryo  {
          idstrategy  = "default"
        }
      }
    }
    """)
    val system = ActorSystem("testSystem", defaultConfig)
    val serialization = SerializationExtension(system)

    val obuf1 = new Output(1024, 1024 * 1024)

    val listOfBars = Time.values.map(Bar(_)).toList
    val bytes = serialization.serialize(listOfBars).get

    val futures = 1 to 2 map(_ => Future[List[Bar]] {
      serialization.deserialize(bytes.clone, classOf[List[Bar]]).get
    })

    val result = Await.result(Future.sequence(futures), Duration.Inf)
    assert(result.map(_.map(_.time)).flatten.distinct.size == Time.values.size)
  }
```

Output:

```
Vector(Second, Month, Minute, Day, Year, Hour, Second, null) had size 8 instead of expected size 6
```

### PR DESCRIPTION
**Pull request by luben**
Another shot, I think I got it correctly this time, fixes #75, added also a test.

### CODE REVIEW

**Comment by romix** [comment]
I'm concerned about any potential performance issues of your approach. You now perform the reflection operation "getDeclaredField" on each call. This is usually a very bad idea....

**Comment by luben** [comment]
@romix you can pull only the first commit, it uses caches for both reads and writes but I do not see any performance difference - I was using the benchmark in the new EnumTest for the performance evaluation.

**Comment by luben** [comment]
@romix do you want to scrap this PR and send another one with cached reflective lookups?

**Comment by romix** [comment]
@luben Sorry, I was pretty busy at work. I think we can merge as is and if someone will complain about performance, we can introduce caching later.

File: src/test/scala/com/romix/scala/serialization/kryo/EnumerationSerializationTest.scala [hunk 1]
```diff
@@ -29,7 +29,7 @@ class EnumarationSerializationTest extends SpecCase {
     val bytes = obuf1.toBytes
     val ibuf1 = new Input(bytes)
     val enumObjWeekday1 = kryo.readClassAndObject(ibuf1)
-    val enumObjTime1 = kryo.readClassAndObject(ibuf1)
+    val enumObjTime1    = kryo.readClassAndObject(ibuf1)
```
Comment (romix) [inline src/test/scala/com/romix/scala/serialization/kryo/EnumerationSerializationTest.scala]: Why do we have these whitespaces or tab here?

**Comment by luben** [comment]
OK, yes, my tests on JVM8 showed no difference in performance but it may depend on other factors.

**Comment by romix** [comment]
Pleas fix the space/tab issue and I'll merge.

File: src/test/scala/com/romix/scala/serialization/kryo/EnumerationSerializationTest.scala (hunk 1 above)
Comment (luben) [inline src/test/scala/com/romix/scala/serialization/kryo/EnumerationSerializationTest.scala]: unintended, vertical aliged it with the assignment on the previous line

**Comment by romix** [comment]
And may be squash the whole thing in one commit?

**Comment by luben** [comment]
@romix all squashed in one commit

**Comment by romix** [comment]
Thanks!

**Pull request merged**